### PR TITLE
Make libvirt-coreos generate a .kubeconfig file and have e2e tests use it

### PR DESCRIPTION
**Pull request by lhuard1A**
Since commit 7db006ab it is mandatory to have either a `kubeconfig` or an `auth_config` passed as parameter to e2e
Without it, e2e issues a bunch of errors:

```
  Expected error:
      <*errors.errorString | 0xc208175230>: {
          s: "error creating client: either kubeConfig or authConfig must be specified to load client config",
      }
      error creating client: either kubeConfig or authConfig must be specified to load client config
  not to have occurred
```

This PR fixes the above mentioned errors on libvirt-coreos cluster.

### CODE REVIEW

File: cluster/libvirt-coreos/util.sh [hunk 1]
```diff
@@ -220,6 +220,13 @@ function kube-up {
     rm $domain_xml
   done
 
+  export KUBECONFIG="$ROOT/.kubeconfig"
```
Comment (ixdy) [inline cluster/libvirt-coreos/util.sh]: In cluster/gce/util.sh this is set to ${HOME}/.kube/.kubeconfig. Why did you opt to put it under cluster/libvert-coreos here?

File: hack/ginkgo-e2e.sh [hunk 2]
```diff
@@ -130,13 +128,12 @@ else
...
     "--cert_dir=${CERT_DIR:-}"
   )
-  host="https://${KUBE_MASTER_IP-}"
 fi
 
 # Use the kubectl binary from the same directory as the e2e binary.
 export PATH=$(dirname "${e2e}"):"${PATH}"
 "${e2e}" "${auth_config[@]:+${auth_config[@]}}" \
-  --host="$host" \
+  --host="https://${KUBE_MASTER_IP-}" \
```
Comment (ixdy) [inline hack/ginkgo-e2e.sh]: In cluster/libvirt-coreos/util.sh you use http://${KUBE_MASTER_IP-}:8080 which is now inconsistent with what's used here (and was used here previously) - was this change intentional?

File: cluster/libvirt-coreos/util.sh (hunk 1 above)
Comment (lhuard1A) [inline cluster/libvirt-coreos/util.sh]: I have amended the PR in order to reuse ${HOME}/.kube/.kubeconfig .
It new reuses the same config file as GCE but with a dedicated cluster name.

File: hack/ginkgo-e2e.sh (hunk 2 above)
Comment (lhuard1A) [inline hack/ginkgo-e2e.sh]: Indeed, it becomes inconsistent but, when using a `.kubeconfig` file, the host to connect to is found in the `.kubeconfig` file and the `--host` parameter is ignored.

The code is there:
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/test/e2e/util.go#L120
* If we provide a `.kubeconfig` everything comes from `testContext.kubeConfig`.
* If we provide a `--auth_config`, then we use `testContext.host`.

The goal of my change was to remove the libvirt-coreos specific setting of `--host` because it is not used anymore.

In order to remove that misleading trick, we could move the `--host` parameter to the provider specific `auth_config` section, except for the ones using `kubeconfig`?

Comment (ixdy) [inline hack/ginkgo-e2e.sh]: Yeah, it might make the code here a little clearer if --host were only provided when it was needed.
OTOH, that means duplicating exactly the same line 5 times, which also isn't great.
Maybe add a comment or something that mentions when --auth_config is used, --host is ignored?

Comment (lhuard1A) [inline hack/ginkgo-e2e.sh]: Yes, and checking whether `auth_config` contains `--auth_config` or `--kubeconfig` in order to add `--host` only in the former case would make things uselessly more complex.

I just added a comment to stress the fact that `--host` is not always used. Could you check if it’s OK for you?
